File: linkedlistdemo.java
import java.util.*;
import java.lang.*;
import java.io.*;
class node{
	int data;
	node next;
	node(int n){
		data=n;
		next=null;
	}
}
class Ideone
{
	void display(node temp)
	{
		while(temp!=null){
			System.out.print(temp.data+"\t");
			temp=temp.next;
		}
	}
	public static void main (String[] args)
	{
		node head,one,two,three;
		head=new node(1);
		one=new node(2);
		two=new node(3);
		three=new node(4);
		head.next=one;
		one.next=two;
		two.next=three;
		three.next=null;
		Ideone Ob=new Ideone();
		Ob.display(head);
	}
}

File: linkedlistdemo.c
#include <stdio.h>
#include <stdlib.h>
struct node{
	int data;
	struct node *next;
	};
typedef struct node nd;
void display(nd *temp)
{
	while(temp!=NULL){
		printf("%d\t",temp->data);
		temp=temp->next;
	}
}
int main(void) {
	nd *head,*one,*two,*three;
	head=(nd*)malloc(sizeof(nd));
	one=(nd*)malloc(sizeof(nd));
	two=(nd*)malloc(sizeof(nd));
	three=(nd*)malloc(sizeof(nd));
	head->data=1;
	head->next=one;
	one->data=2;
	one->next=two;
	two->data=3;
	two->next=three;
	three->data=4;
	three->next=NULL;
	display(head);
	return 0;
}
